letsencrypt/cli.py

```diff
@@ -283,22 +282,58 @@ def run(args, config, plugins):  # pylint: disable=too-many-branches,too-many-lo
         lineage.update_all_links_to(lineage.latest_common_version())
         # TODO: Check return value of save_successor
         # TODO: Also update lineage renewal config with any relevant
-        #       configuration values from this attempt?
-        le_client.deploy_certificate(
-            domains, lineage.privkey, lineage.cert, lineage.chain)
-        display_ops.success_renewal(domains)
+        #       configuration values from this attempt? - YES
     else:
         # TREAT AS NEW REQUEST
         lineage = le_client.obtain_and_enroll_certificate(
-            domains, authenticator, installer, plugins)
+            domains, le_client.dv_auth, le_client.installer, plugins)
         if not lineage:
-            return "Certificate could not be obtained"
-        # TODO: This treats the key as changed even when it wasn't
```

Comment:
My understanding was that this TODO was incorrect.  You must always deploy the key, even if it hasn't changed.  

The underlying configuration could have changed.  There could be new vhosts that now use the cert/key but they don't have them yet.  I don't think it should matter if the key has changed or not... assumptions shouldn't be made about the current underlying configuration.

At the same time... if it is the same key... the configurations won't be updated anyway. (but they will have the opportunity to update if they need to)